alphacsc/learn_d_z_multi.py

```diff
@@ -295,7 +297,7 @@ def _batch_learn(z_encoder, d_solver, end_iter_func, n_iter=100,
         if verbose > 5:
             print(
                 '[{}] Objective (z) : {:.3e} (sparsity: {:.3e})'
-                .format(name, pobj[-1], z_nnz.mean())
+                .format(name, pobj[-1], (z_nnz/z_nnz.shape[-1]).mean())
```

Comment:
I don't think this is the correct normalization.
This should be :
```suggestion
                .format(name, pobj[-1], (z_nn.mean() / np.prod(z.shape))
```
Which will give the sparsity ratio.

Reply:
> 

```suggestion
                .format(name, pobj[-1], (z_nnz.sum() / np.prod(z.shape))
```
Would not it be a sum rather than a mean?